contracts/poap/src/contract_tests.rs

```diff
@@ -0,0 +1,962 @@
+            poap_code_id,
+            Addr::unchecked(ADMIN),
+            &init_msg,
+            &[],
+            "Poap contract",
+            None,
+        );
+
+        assert!(init_result.is_err());
+    }
+
+    #[test]
+    fn instantiate_with_event_start_eq_end() {
+        let mut app = mock_app();
+        let (cw721_code_id, poap_code_id) = store_contracts(&mut app);
+        let mut init_msg = get_valid_init_msg(cw721_code_id);
+
+        let current_block = app.block_info();
+        // Create a start time 200 seconds after the current block time
+        let start = current_block.time.seconds() + 200;
+        init_msg.event_info.start_time = Timestamp::from_seconds(start);
+        init_msg.event_info.end_time = Timestamp::from_seconds(start);
+
+        let init_result = app.instantiate_contract(
+            poap_code_id,
+            Addr::unchecked(ADMIN),
+            &init_msg,
+            &[],
+            "Poap contract",
+            None,
+        );
+
+        assert!(init_result.is_err());
+    }
+
+    #[test]
+    fn instantiate_with_invalid_poap_uri() {
+        let mut app = mock_app();
+        let (cw721_code_id, poap_code_id) = store_contracts(&mut app);
+        let mut init_msg = get_valid_init_msg(cw721_code_id);
+
+        // Invalid uri
+        init_msg.event_info.base_poap_uri = "invalid_uri".to_string();
+        let init_result = app.instantiate_contract(
+            poap_code_id,
+            Addr::unchecked(ADMIN),
+            &init_msg,
+            &[],
+            "Poap contract",
+            None,
+        );
+        assert!(init_result.is_err());
+
+        // Non ipfs uri
+        init_msg.event_info.base_poap_uri = "https://random_domain.com".to_string();
+        let init_result = app.instantiate_contract(
+            poap_code_id,
+            Addr::unchecked(ADMIN),
+            &init_msg,
+            &[],
+            "Poap contract",
+            None,
+        );
+        assert!(init_result.is_err());
+    }
+
+    #[test]
+    fn instantiate_with_invalid_event_uri() {
+        let mut app = mock_app();
+        let (cw721_code_id, poap_code_id) = store_contracts(&mut app);
+        let mut init_msg = get_valid_init_msg(cw721_code_id);
+
+        // Invalid uri
+        init_msg.event_info.event_uri = "invalid_uri".to_string();
+        let init_result = app.instantiate_contract(
+            poap_code_id,
+            Addr::unchecked(ADMIN),
+            &init_msg,
+            &[],
+            "Poap contract",
+            None,
+        );
+        assert!(init_result.is_err());
+
+        // Non ipfs uri
+        init_msg.event_info.event_uri = "https://random_domain.com".to_string();
+        let init_result = app.instantiate_contract(
+            poap_code_id,
+            Addr::unchecked(ADMIN),
+            &init_msg,
+            &[],
+            "Poap contract",
+            None,
+        );
+        assert!(init_result.is_err());
+    }
+
```

Comment:
Missing a test for `proper_instantiate` with `config`, `event_info` and etc. state check.

Reply:
`proper_instantiate` is an utility that provides a correclty initialized enviroement. If the initialization fails the test fail providing the reason and a stack trace since is using `unwrap` [here](https://github.com/desmos-labs/desmos-contracts/blob/909561e5c50798c81651c68c07a823b5f8a1a111/contracts/poap/src/contract_tests.rs#L105).
The tests to check the initialization with invalid params are covered in the `instantiate_...` tests.

Reply:
Yes, `unwrap` check the initialization fails, but the state check on cw721 is missing. Say, we should check the cw721 `minter` is poap contract  after initialization.

Reply:
Totally forgot that we override the minter address during initialization, nice catch!
I will add a test that after calling `proper_instantiate` checks the cw721 and poap contract state.